pyogrio/_io.pyx

```diff
@@ -611,6 +611,10 @@ cdef detect_encoding(OGRDataSourceH ogr_dataset, OGRLayerH ogr_layer):
         # In old gdal versions, OLCStringsAsUTF8 wasn't advertised yet.
         return "UTF-8"
 
+    if driver == "SQLite":
+        # In gdal versions < 3.12, OLCStringsAsUTF8 wasn't advertised yet.
```

Comment:
Is this fixed in GDAL as of 3.12?   I didn't see a fix referenced in your upstream GDAL issue.  Maybe for now just put in the link to https://github.com/OSGeo/gdal/issues/12962 and later we'll update that to the correct version and add a version check around this.

Reply:
Indeed, I was a bit optimistic :-).